src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+
+        final String prefix = FilenameUtils.getBaseName(inputFASTQFile.getName());//extractBaseNameWithoutExtension(inputFASTQFile);
+        final String preprocessedFASTAFileName = prefix+".pp.fa";
+
+        final SGAModule preprocess = new SGAModule("preprocess");
+        final ArrayList<String> ppArgs = new ArrayList<>();
+        ppArgs.add("--pe-mode");    ppArgs.add("2");
+        ppArgs.add("--pe-orphans"); ppArgs.add(prefix+".pp.orphan.fa");
+        ppArgs.add("--out");        ppArgs.add(preprocessedFASTAFileName);
+        ppArgs.add(inputFASTQFile.getName());
+
+        final Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String> ppInfo = preprocess.run(sgaPath, outputDirectory, ppArgs);
+
+        indexerArgs.set(indexerArgs.size()-1, preprocessedFASTAFileName);
+        final Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String> indexerInfo = indexer.run(sgaPath, outputDirectory, indexerArgs);
+
+        final SGAUnitRuntimeInfoStruct ppUnitInfo = new SGAUnitRuntimeInfoStruct("preprocess", preprocessedFASTAFileName, ppInfo._1(), ppInfo._2(), ppInfo._3());
+        ppUnitInfo.add("index", indexerInfo._1(), indexerInfo._2(), indexerInfo._3());
+
+        return ppUnitInfo;
+    }
+
+    @VisibleForTesting
+    static SGAUnitRuntimeInfoStruct runSGACorrect(final Path sgaPath,
+                                                  final File inputFASTAFile,
+                                                  final File outputDirectory,
+                                                  final int threads,
+                                                  final SGAModule indexer,
+                                                  final ArrayList<String> indexerArgs){
+        return runSimpleModuleFollowedByIndexing(sgaPath, "correct", ".ec.fa", inputFASTAFile, outputDirectory, threads, indexer, indexerArgs);
+    }
+
+    @VisibleForTesting
+    static SGAUnitRuntimeInfoStruct runSGAFilter(final Path sgaPath,
+                                                 final File inputFASTAFile,
+                                                 final File outputDirectory,
+                                                 final int threads){
+        final String prefix = FilenameUtils.getBaseName(inputFASTAFile.getName());//extractBaseNameWithoutExtension(inputFASTAFile);
+        final SGAModule filter = new SGAModule("filter");
+        final ArrayList<String> filterArgs = new ArrayList<>();
+        filterArgs.add("--threads"); filterArgs.add(Integer.toString(threads));
+        filterArgs.add(prefix+".fa");
+        final Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String> filterInfo = filter.run(sgaPath, outputDirectory, filterArgs);
+
+        return new SGAUnitRuntimeInfoStruct("filer", prefix+".filter.pass.fa", filterInfo._1(), filterInfo._2(), filterInfo._3());
+    }
+
+    @VisibleForTesting
+    static SGAUnitRuntimeInfoStruct runSGARmDuplicate(final Path sgaPath,
```

Comment:
In general I think the reads we are getting from the breakpoint finding tool are already duplicate filtered (based on mapping location). Have you experimented with whether this step is necessary?